source/base/NexusApp.cc

```diff
@@ -105,13 +105,13 @@ NexusApp::NexusApp(G4String init_macro): G4RunManager(), gen_name_(""),
   pg->SetGenerator(ObjFactory<G4VPrimaryGenerator>::Instance().CreateObject(gen_name_));
   this->SetUserAction(pg.release());
 
-  if (pm_name_ == "") {
-    G4Exception("[NexusApp]", "NexusApp()", FatalException, "A persistency manager must be specified.");
-  }
-  pm_ = ObjFactory<PersistencyManagerBase>::Instance().CreateObject(pm_name_);
-  pm_->SetMacros(init_macro, macros_, delayed_);
 
- // PersistencyManager::Initialize(init_macro, macros_, delayed_);
+  // Set the persistency manager, if needed
+  if (pm_name_ != "") {
```

Comment:
I think `!pm_name.empty()` is slightly faster than `!= ""` and the "recommended" way